tests/core/test_memory.py

```diff
@@ -186,36 +238,92 @@ def test_summary_parameter_count(mode):
     ]
 
 
-@pytest.mark.parametrize(['mode'], [
-    pytest.param(ModelSummary.MODE_FULL),
-    pytest.param(ModelSummary.MODE_TOP),
-])
+@pytest.mark.parametrize(
+    ["mode"],
+    [
+        pytest.param(ModelSummary.MODE_FULL),
+        pytest.param(ModelSummary.MODE_TOP),
+    ],
+)
 def test_summary_layer_types(mode):
     """ Test that the summary displays the layer names correctly. """
     model = UnorderedModel()
     summary = model.summarize(mode=mode)
     assert summary.layer_types == [
-        'Linear',
-        'Linear',
-        'Linear',
-        'ReLU',
-        'Conv2d',
+        "Linear",
+        "Linear",
+        "Linear",
+        "ReLU",
+        "Conv2d",
     ]
 
 
-@pytest.mark.parametrize(['mode'], [
-    pytest.param(ModelSummary.MODE_FULL),
-    pytest.param(ModelSummary.MODE_TOP),
-])
-@pytest.mark.parametrize(['example_input', 'expected_size'], [
-    pytest.param([], UNKNOWN_SIZE),
-    pytest.param((1, 2, 3), [UNKNOWN_SIZE] * 3),
-    pytest.param(torch.tensor(0), UNKNOWN_SIZE),
-    pytest.param(dict(tensor=torch.zeros(1, 2, 3)), UNKNOWN_SIZE),
-    pytest.param(torch.zeros(2, 3, 4), [2, 3, 4]),
-    pytest.param([torch.zeros(2, 3), torch.zeros(4, 5)], [[2, 3], [4, 5]]),
-    pytest.param((torch.zeros(2, 3), torch.zeros(4, 5)), [[2, 3], [4, 5]]),
-])
+@pytest.mark.parametrize(
+    ["mode"],
+    [
+        pytest.param(ModelSummary.MODE_FULL),
+        pytest.param(ModelSummary.MODE_TOP),
+    ],
+)
+@pytest.mark.parametrize(
+    ["example_input", "expected_model_size"],
+    [
+        pytest.param(torch.zeros(1, 1, 28, 28), 0.318),
+        pytest.param(torch.zeros(1, 1, 224, 224), 31.84),
+        pytest.param(torch.zeros(10, 1, 512, 512), 183.425),
+        pytest.param(None, 0.075),
+    ],
+)
+def test_known_model_sizes(example_input, expected_model_size, mode):
+    """ Test the knownet model on example input arrays and corresponding known model size """
+
+    model = KnownNet()
+    model.example_input_array = example_input
+    summary = model.summarize(mode=mode)
+    assert almost_equals(summary.model_size(), expected_model_size, rel_tol=1e-3, abs_tol=1e-3)
+
+
+@pytest.mark.parametrize(
+    ["mode"],
+    [
+        pytest.param(ModelSummary.MODE_FULL),
+    ],
+)
```

Comment:
We don't need this parametrize if there is only one case, you can set the mode in the function

Reply:
Yeah but just for consistency in test functions i did this
I can explicitly initialize summary with mode too :) should I?

Reply:
Sure, you are only testing for that single case here.

Reply:
ok